cmd/logmonitor/logmonitor.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian/client"
+	"github.com/jinzhu/gorm"
+)
+
+// LogMonitor is an object used to continuously monitor a Trillian log.
+type LogMonitor struct {
+	logID                int64
+	logClient            *client.LogClient
+	db                   *gorm.DB
+	waitBetweenDBUpdates time.Duration
+}
+
+// Run starts continuously monitoring the Trillian log specified in the
+// LogMonitor object.
+//
+// It continues running until it encounters an error.
+func (m *LogMonitor) Run(ctx context.Context) error {
```

Comment:
I started doing this, but then realized it would add a bit of complexity and boilerplate since the majority of the inner loop logic is about handling breaking out of the infinite loop.

Since a single-shot is the same as running `LogClient.WaitForRootUpdate`, it makes sense to leave single-shot testing to LogClient tests, no?

Reply:
By splitting it out into a separate method, I was hoping that we could eliminate a bunch of that complexity.